api/sixtyfps-cpp/include/sixtyfps.h

```diff
@@ -431,11 +431,17 @@ class Model
     /// Returns the data for a particular row. This function should be called with `row <
     /// row_count()`.
     virtual ModelData row_data(int i) const = 0;
-    /// Sets the data for a particular row. This function should be called with `row < row_count()`.
-    /// If the model cannot support data changes, then it is ok to do nothing (default
-    /// implementation). If the model can update the data, the implementation should also call
-    /// row_changed.
-    virtual void set_row_data(int, const ModelData &) {};
+    /// Sets the data for a particular row.
+    ///
+    /// This function should only be called with `row < row_count()`.
+    ///
+    /// If the model cannot support data changes, then it is ok to do nothing.
+    /// The default implementation will print a warning to stderr.
```

Comment:
We could also consider using a return value to indicate to the caller that the insertion/mutation failed. But that's a breaking change, not for now :)

Reply:
But then who would print report the error?  The generated code that call that function?

Reply:
Yes, when the caller is generated code (or we'll collect it in general binding error code that may need in the future). If the caller is the user, then they have a way of acting. Could even be annotated with a `__attribute__ ((__warn_unused_result__))`.